FreeSurfer/custom/SetUpFSNHP.sh

```diff
@@ -0,0 +1,268 @@
+#! /bin/bash
+
+# Species specific config file for recon-all.v6.hiresNHP
+
+# Takuya Hayashi, RIKEN Japan 
+# Akiko Uematsu, RIKEN Japan 
+# Chad Donahue, Washington University St. Louis, USA
+# Matthew F Glasser, Washington University St. Louis, USA
+# Copyright (c) 2018-2024
+# All rights reserved.
+
+SkullStripMethod=PreFS                                             # PreFS or FS. PreFS method is recommended for NHP - TH Feb 2024
+
+
+if [[ "$SPECIES" =~ Human ]] ; then
+
+  IntensityCor="FAST"
+  TemplateWMSkeleton="NONE"
+  ScaleFactor=1
+  mri_segment_args=""
+  mri_fill_args="-C 0 0 27"
+  T2normSigma=""
+  VariableSigma=""
+  PialSigma=""
+  SmoothNiter=""
+  NSigmaAbove=""
+  NSigmaBelow=""    
+  WMProjAbs="2"
+  VariableSigmaFS5="8"                                        # FS5.3.0 HighResPial  
+  MaxThickness="6"                                            # FS5.3.0 HighResPial   
+  GreySigma="5"                                               # FS5.3.0 HighResPial 
+  BiasFieldFastSmoothingSigma=""  # = 20*$ScaleFactor
+  GCAdir="${FREESURFER_HOME}/average"
+  GCA="RB_all_2016-05-10.vc700.gca"
+  AvgCurvTif="average.curvature.filled.buckner40.tif"
+
+  # FS BBR in DiffusionPreprocPipeline
+  FSBBRDIFF=TRUE
+  DiffWMProjAbs="2"
+
+elif [[ "$SPECIES" =~ Chimp ]] ; then
+
+  IntensityCor="FAST"
+  TemplateWMSkeleton="NONE"
+  ScaleFactor="1.25"
+  BiasFieldFastSmoothingSigma="25"                            # = 20*ScaleFactor
+  StrongBias_args="-s"                                        # strong bias in T1w
+  mri_normalize_args="-sigma 10 -gentle"                      # = 8*ScaleFactor.
+  mri_segment_args=""    
+  initcc="128 110 125"                                        # FS5.3.0 voxel coordinate in orig.mgz and wm.mgz
+  mri_fill_args="-C 0 -8 21"                                  # mm coordinate in orig.mgz and wm.mgz
+  mris_inflate1_args="-n 250"                                  # set higher # of iterations (default=10) to avoid "white surface short cut"
+  T1normSigma=""
+  T2normSigma="20"
+  VariableSigma="3"
+  PialSigma="2"
+  SmoothNiter="1"
+  if [ "${p_flair}" != "TRUE" ] ; then                        # control T2 pial
```

Comment:
Don't use legacy `opts.shlib` parsing details, pass in what you need to know as an argument.
```suggestion
  if ((isFLAIR)) ; then                        # control T2 pial
```
This particular conditional currently does the same thing on both branches, though.

Reply:
Sorry, I have applied the same fix to Marmoset and the other modules as well.
916f3e778bddaa87f1817a224911a76c89105c81